controlpanel/api/aws.py

```diff
@@ -221,32 +224,31 @@ class S3AccessPolicy:
 
     def __init__(self, policy):
         self.policy = policy
-        self._policy_document = None
         self.statements = {}
 
-        self._load()
-
-    def _load(self):
         try:
-            # ensure version is set
-            self.policy_document['Version'] = '2012-10-17'
+            policy_document = self.load_policy_document()
 
         except self.policy.meta.client.exceptions.NoSuchEntityException:
             # create an empty s3 access policy
-            self.put(policy_document=deepcopy(BASE_S3_ACCESS_POLICY))
+            policy_document = deepcopy(BASE_S3_ACCESS_POLICY)
+            self.put(policy_document=policy_document)
+
+        # ensure version is set
+        policy_document['Version'] = '2012-10-17'
 
         # ensure statements are correctly configured and build a lookup table
-        for stmt in self.policy_document.setdefault('Statement', []):
+        for stmt in policy_document.setdefault('Statement', []):
             sid = stmt.get('Sid')
             if sid in ('list', 'readonly', 'readwrite'):
-                stmt.update(BASE_S3_ACCESS_STATEMENT[sid])
+                stmt.update(deepcopy(BASE_S3_ACCESS_STATEMENT[sid]))
                 self.statements[sid] = stmt
 
-    @property
-    def policy_document(self):
-        if self._policy_document is None:
-            self._policy_document = self.policy.policy_document
-        return self._policy_document
+        self.policy_document = policy_document
+
+    def load_policy_document(self):
```

Comment:
Do we need to add a method? Is it used anywhere? If not I reckon could be simpler to inline in the constructor (but you know the code better)

Reply:
This is to allow the subclass to override it without having to copy out the rest of the constructor code.